### PR TITLE
use crs widget picker in db_manager

### PR DESCRIPTION
**Pull request by fanevanjanahary**
## Description

* Use CRS widget instead of QLineEdit for CRS in vector import dialog
* Review tab order in this dialog

Before:
before

After:
after

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by slarosa** [comment]
@Fanevanjanahary thank you very much for this very useful improvements. Very appreciated! 👍

File: python/plugins/db_manager/dlg_import_vector.py [hunk 1]
```diff
@@ -68,6 +68,8 @@ def __init__(self, inLayer, outDb, outUri, parent=None):
         # updates of UI
         self.setupWorkingMode(self.mode)
         self.cboSchema.currentIndexChanged.connect(self.populateTables)
+        self.widgetSourceSrid.setOptionVisible(QgsProjectionSelectionWidget.CrsNotSet, True)
```
Comment (nyalldawson) [inline python/plugins/db_manager/dlg_import_vector.py]: I'm wondering - do we need to show the "not set" options? later we just show an error if they are used.

**Review by DelazJ — COMMENTED** [review]
Added some comments relative to HIG guidelines. 
Merci.

File: python/plugins/db_manager/dlg_import_vector.py (hunk 1 above)
Comment (Fanevanjanahary) [reply python/plugins/db_manager/dlg_import_vector.py]: I agree but I did this to hide the "invalid projection" when the check-box is unchecked

**Comment by Fanevanjanahary** [comment]
Thank you for review, I fixed HIG

**Comment by rduivenvoorde** [comment]
Nice first contribution!! Thanks!

**Comment by Gustry** [comment]
Nice thanks @Fanevanjanahary !

Can this land in 3.0? It looks like just replacing the old QLineEdit with the CRS widget.

File: python/plugins/db_manager/dlg_import_vector.py (hunk 1 above)
Comment (nyalldawson) [reply python/plugins/db_manager/dlg_import_vector.py]: Hm - I just tested this branch, and at the moment this is what I see:

1. open the dialog - crs options are blank and corresponding checkboxes are unticked
2. tick a checkbox - the crs option is still blank

This seems odd to me. If the checkbox is ticked then we should be forcing a choice in the crs selector, not giving an empty crs.

I'd suggest:
- default to project CRS as the initial (disabled) choice
- if a layer from the project is picked from the selector, reset the widgets to match that layer's projection

Comment (Fanevanjanahary) [reply python/plugins/db_manager/dlg_import_vector.py]: I understand, I changed to show the project CRS by default. For layer's projection, the source code is bit confusing. Can we let like this? It's better than before.

**Comment by nirvn** [comment]
Is this ready to be merged?

**Comment by Fanevanjanahary** [comment]
Hi @nirvn, yes I would like it to be merged if possible. Thanks

**Comment by m-kuhn** [comment]
Thanks a lot @Fanevanjanahary this looks really good :+1:

**Pull request merged**